report/26_Process-Development-Process-&-Tools.md

```diff
@@ -1,2 +1,52 @@
+In this, we can attach all relevant issues, set a due date and afterward track the combined progress for all tasks.
+When a developer is ready to solve the issue, they first assign themself if they were not assigned when the issue was created. Afterward, they make a branch and make the changes to the code, as described in (TODO: Refer to GitHub flow chapter). When the code change is complete, they create a new PR from their branch to master, and augment it with the relevant data, such as a descriptive title, description, and linked issues. By creating this process, it signals to the other developers that the code is ready for review. Furthermore, the developer also posts a link to the PR on discord to notify the rest of the team.
+
+For a pull request to be merged it must comply with our policies.
+First of all, all automatic tests and checks must complete successfully. This includes unit tests, end-to-end tests, static analysis, and more.
+Secondly, the PR must be approved by at least one other developer. Github supports code reviews as a part of the pull requests, so that other team members can, approve, reject, and suggest changes to a PR.
+The automatic tests help catch bugs and regressions the human reviewer would otherwise miss, and visa versa.
+Even though we cannot eliminate bugs completely, but the combination of automatic tools, combined with human common sense, is the best way of attempting to prevent them.
+If a pull request is rejected, the author can make changes and re-request reviews.
+When a pull request is merged, it will automatically be deployed to our production environment (TODO: Refer to CI/CD chapter), and mark all linked issues as solved.
+<!-- 
+**TODO GitHub**:
+* Issues
+* PR's
+* Reviews
+* Labels
+* Milestones 
+-->
+
+Discord has been used where GitHub falls short, namely for instant synchronous communication.
+Despite the "gamer-branding" we used Discord instead of other popular team communication tools, such as messenger or slack, for various reasons.
+Besides already using it extensively in our daily lives, Discord has a powerful feature set that makes it incredibly well suited for distributed teamwork.
+One of the biggest advantages is that a discord server supports multiple text and voice channels. 
+This makes it easy to categorize and organize the chat. The voice chats also allow us to enter and leave at will, making it ideal for long-term group work, where people can join and leave as they please.
+Another big advantage is that discord has an easy-to-use API, with support for webhooks. This has been utilized by hooking up GitHub and our uptime-monitoring tool, OhDear, to separate channels, so that we get automatically get notified when new updates occur. This has allowed us to act very fast on code changes and downtime.
+
+
+
+<!-- 
+**TODO Discord**
+
+* Channels
+* Voice chats
+* Bots 
+-->
+
+**TODO: Descripe how we where devops when writing report**
```

Comment:
Dont know if this actually fits here or somewhere else

Reply:
It fits here.

Reply:
I have no idea what that sentence means